### PR TITLE
Созонов Илья. Задача 1. Вариант 7. Нахождение наиболее близких соседних элементов вектора.

### PR DESCRIPTION
**Pull request by sozozzya**
**Описание последовательной задачи:**
1. Создаём переменную min и инициализируем её максимальным возможным значением INT_MAX. 
2. Проходим по вектору, начиная с первого элемента до предпоследнего, и находим модуль разности между текущим элементом и следующим за ним. 
3. Если модуль разности между ними меньше текущего значения min, обновляем min и запоминаем индекс текущего элемента вектора. 

**Описание MPI задачи:**
1. Создаем новый вектор diff, состоящий из пар: модуль разности соседних элементов исходного вектора и соответствующего индекса. 
2. Полученный вектор пар делится равномерно на сегменты по числу процессов. 
3. Каждый сегмент данных отправляется от нулевого процесса остальным. 
5. Каждый процесс находит наименьший элемент и соответствующий ему индекс в полученном сегменте данных. 
6. После завершения локальных вычислений процессы возвращают результаты корневому процессу, где с помощью коллективной операции reduce находится наименьший модуль разности соседних элементов вектора и соответствующий индекс.

### CODE REVIEW

File: tasks/mpi/sozonov_i_nearest_neighbor_elements/src/ops_mpi.cpp [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+}
+
+bool sozonov_i_nearest_neighbor_elements_mpi::TestMPITaskSequential::post_processing() {
+  internal_order_test();
+  reinterpret_cast<int*>(taskData->outputs[0])[0] = input_[res];
+  reinterpret_cast<int*>(taskData->outputs[0])[1] = input_[res + 1];
+  return true;
+}
+
+bool sozonov_i_nearest_neighbor_elements_mpi::TestMPITaskParallel::pre_processing() {
```
Comment (allnes) [inline tasks/mpi/sozonov_i_nearest_neighbor_elements/src/ops_mpi.cpp:60]: please use all mpi functions in run()

**Review by allnes — CHANGES_REQUESTED** [review]

File: tasks/mpi/sozonov_i_nearest_neighbor_elements/func_tests/main.cpp [hunk 2]
```diff
@@ -0,0 +1,183 @@
+#include <gtest/gtest.h>
```
Comment (KolodkinGrigorii) [inline tasks/mpi/sozonov_i_nearest_neighbor_elements/func_tests/main.cpp:1]: add test for empty vector
